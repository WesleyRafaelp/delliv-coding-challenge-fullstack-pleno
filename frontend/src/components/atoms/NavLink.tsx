import React from 'react';
import { useRouter } from 'next/router';

interface NavLinkProps {
  href: string;
  children: React.ReactNode;
}

const NavLink: React.FC<NavLinkProps> = ({ href, children }) => {
  const router = useRouter();

  const isActive = router.pathname === href;

  return (
    <a href={href} className={`text-white-700
        hover:text-blue-500
        ${isActive ? 'border-b-2 border-blue-500' : 'border-b-2 border-transparent'}
        transition-all duration-300 ease-in-out
        cursor-pointer
        font-semibold
        text-xl
      `}>
      {children}
    </a>
  );
};

export default NavLink;
