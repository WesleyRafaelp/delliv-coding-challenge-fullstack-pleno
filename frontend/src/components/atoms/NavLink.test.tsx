import React from 'react';
import { render, screen } from '@testing-library/react';
import NavLink from './NavLink';
import 'tailwindcss'

jest.mock('next/link', () => ({ children }: any) => children);

jest.mock('next/router', () => ({
  useRouter: () => ({
    pathname: '/',
    push: jest.fn(),
  }),
}));

describe('NavLink Component', () => {
  it('renders NavLink with specified href and children', () => {
    render(<NavLink href="/sample-path">Sample NavLink</NavLink>);

    const linkElement = screen.getByRole('link', { name: /Sample NavLink/i });

    expect(linkElement).toHaveAttribute('href', '/sample-path');
    expect(linkElement).toHaveClass('border-b-2', 'border-transparent');
  });
  

  it('adds active styles when the link is active', () => {
    jest.spyOn(require('next/router'), 'useRouter').mockReturnValue({
      pathname: '/active-path',
      push: jest.fn(),
    });
  
    render(<NavLink href="/active-path">Active NavLink</NavLink>);
  
    const navLinkElement = screen.getByText(/Active NavLink/i);
    
    
    expect(navLinkElement).toHaveClass('border-b-2');
    
   expect(navLinkElement).toHaveClass('border-blue-500');
    
    expect(navLinkElement).not.toHaveClass('border-transparent');
  });
  

  it('does not add active styles when the link is not active', () => {
    jest.spyOn(require('next/router'), 'useRouter').mockReturnValue({
      pathname: '/inactive-path',
      push: jest.fn(),
    });

    render(<NavLink href="/active-path">Inactive NavLink</NavLink>);

    const navLinkElement = screen.getByText(/Inactive NavLink/i);
expect(navLinkElement).not.toHaveClass('border-blue-500'); 
expect(navLinkElement).toHaveClass('border-transparent');

  });

  it('matches snapshot', () => {
    render(<NavLink href="/sample-path">Sample NavLink</NavLink>);
    const tree = screen.getByText(/Sample NavLink/i);
    expect(tree).toMatchSnapshot();
  });
});
