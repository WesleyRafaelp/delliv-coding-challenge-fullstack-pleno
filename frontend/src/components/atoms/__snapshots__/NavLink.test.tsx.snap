// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink Component matches snapshot 1`] = `
<a
  class="text-white-700
        hover:text-blue-500
        border-b-2 border-transparent
        transition-all duration-300 ease-in-out
        cursor-pointer
        font-semibold
        text-xl
      "
  href="/sample-path"
>
  Sample NavLink
</a>
`;
